File: matiasmasso/Mat.NET/DT2/DTOBanner.vb

Imports Newtonsoft.Json
Imports SixLabors.ImageSharp

Public Class DTOBanner
    Inherits DTOBaseGuid

    Property Nom As String
    Property SrcGuid As Guid
    Property SrcType As Srcs
    Property FchFrom As Date
    Property FchTo As Date
    Property NavigateTo As String '<AllowHtml>
    Property ImageUrl As String
    Property Product As DTOProduct
    Property Lang As DTOLang
    <JsonIgnore> Public Property Image As Image
    <JsonIgnore> Public Property Thumbnail As Image

    Public Const BANNERWIDTH As Integer = 640
    Public Const BANNERHEIGHT As Integer = 292
    Public Const THUMBWIDTH As Integer = 105
    Public Const THUMBHEIGHT As Integer = 48

    Public Enum Srcs
        notSet
        product
        news
    End Enum

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub


End Class

File: matiasmasso/Mat.NET/DTO/DTOBanner.cs
using System;

namespace DTO
{
    using Newtonsoft.Json;

    public class DTOBanner : DTOBaseGuid
    {
        public string Nom { get; set; }
        public Guid SrcGuid { get; set; }
        public Srcs SrcType { get; set; }
        public DateTime FchFrom { get; set; }
        public DateTime FchTo { get; set; }
        public DTOUrl NavigateTo { get; set; } // <AllowHtml>
        public DTOProduct Product { get; set; }
        public DTOLang Lang { get; set; }
        [JsonIgnore]
        public Byte[] Image { get; set; }
        [JsonIgnore]
        public Byte[] Thumbnail { get; set; }

        public const int BANNERWIDTH = 640;
        public const int BANNERHEIGHT = 292;
        public const int THUMBWIDTH = 105;
        public const int THUMBHEIGHT = 48;

        public enum Srcs
        {
            notSet,
            product,
            news
        }

        public DTOBanner() : base()
        {
        }

        public DTOBanner(Guid oGuid) : base(oGuid)
        {
        }

        public Boolean isActive()
        {
            bool retval = true;
            if (this.FchFrom > DTO.GlobalVariables.Now())
                retval = false;
            if (this.FchTo != DateTime.MinValue && this.FchTo < DTO.GlobalVariables.Now())
                retval = false;
            return retval;
        }

        public string ImageUrl()
        {
            return MmoUrl.ApiUrl("banner/image", Guid.ToString());
        }

    }
}
